### ISSUE TITLE
Line breaks are allowed in composite emojis

### ISSUE THREAD

**Issue by laurmaedje** [typst/typst#80]
Example:
```
#set page("a4")
#lorem(13) 🏳️‍🌈
```

See also: https://github.com/xi-editor/xi-editor/issues/1322

**Comment by tingerrr** [comment]
I took a look at [LD3] and [LD7], both prohibit breaks in joiner sequences.
However, it seems that we do not filter out these prohibited breaks correctly, in this case being `LineBreak::ZWS`.

In, we only check if for mandatory line breaks, but don't try to find the next non-prohibited line break if this one is no mandatory but prohibited:
https://github.com/typst/typst/blob/d3b62bd02ef92b747b724f344e9b8c7d5668ffbe/crates/typst-library/src/layout/par.rs#L1142-L1151

I think I fixed this, the following document, which produced this:
```typst
#set page(width: 50pt, height: auto)
#h(99%) 🏳️‍🌈
🏳️‍🌈
```
![test](https://github.com/typst/typst/assets/137803093/1044dbf0-bca6-4396-85db-cdd866ce2a9a)

Now produces this:
![test](https://github.com/typst/typst/assets/137803093/8394bcb4-16d7-4fe0-aa31-cd175d97689e)

I'll read the spec a bit more closely to ensure I add other prohibited breaks too, and then open a PR.

[LD3]: https://www.unicode.org/reports/tr14/#LD3
[LD7]: https://www.unicode.org/reports/tr14/#LD7

[LB8a]: https://www.unicode.org/reports/tr14/#LB8a

**Comment by tingerrr** [comment]
After reading the docs for `LineSegmenter` again, it should indeed not return prohibited line breaks (thanks @peng1999), I'll open an issue upstream and PR this as a temporary fix. `typst-library`, `1.2.1` but the changelog for `1.3.1` doesn't mention a fix to this specific issue, so updating will likely not fix it for now.

**Comment by tingerrr** [comment]
The upstream issue was fixed and will presumably be released with 1.5, once we update this we can remove the temporary fix added by #2376.
